_posts/2025-08-04-Hosting-OpenSearch-MCP-server-with-Bedrock-AgentCore.md

```diff
@@ -0,0 +1,269 @@
+---
+layout: post
+title: "Hosting the OpenSearch MCP server with Amazon Bedrock AgentCore"
+authors:
+ - jiapingzeng
+date: 2025-08-04
+categories:
+ - technical-post
+meta_keywords: "MCP, Agentic, Agent, Model context protocol, MCP server, LLM integration, observability agent, Bedrock AgentCore, hosted MCP server"
+meta_description: "Learn how to deploy the OpenSearch MCP server with Amazon Bedrock AgentCore Runtime."
+---
+
+The [OpenSearch MCP server](https://github.com/opensearch-project/opensearch-mcp-server-py) enables AI agents to interact with OpenSearch clusters through the Model Context Protocol (MCP). While you can run the OpenSearch MCP server locally, hosting it on Amazon Bedrock AgentCore Runtime provides a scalable, managed solution that's accessible from anywhere.
+
+In this post, we'll walk you through two approaches to deploying the OpenSearch MCP server on Bedrock AgentCore: using an AWS CloudFormation template for a quick setup and manually configuring it using the AgentCore CLI.
+
+## Prerequisites
+
+Before you begin, ensure that you have the following:
+
+- An OpenSearch cluster
+- Access to one of the supported Bedrock AgentCore Regions: `us-east-1`, `us-west-2`, `eu-central-1`, or `ap-southeast-2`
+
+**Note**: While Bedrock AgentCore is only available in these four AWS Regions, your MCP server can connect to OpenSearch clusters in other Regions over the public internet.
+
+## Method 1: Using a CloudFormation template (available to Amazon OpenSearch Service users)
+
+The fastest way to get started is to use the OpenSearch MCP server CloudFormation template, which automatically provisions all necessary resources.
```

Comment:
I think we should also mention other way to use OpenSearch MCP server like deploy the MCP server on EC2?

Reply:
Sounds good